presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControlRules.java

```diff
@@ -19,29 +19,28 @@
 
 import java.util.List;
 import java.util.Optional;
-import java.util.regex.Pattern;
 
 public class FileBasedSystemAccessControlRules
 {
     private final List<CatalogAccessControlRule> catalogRules;
-    private final List<Pattern> userPatterns;
+    private final List<PrincipalUserMatchRule> principalUserMatchRules;
 
     @JsonCreator
     public FileBasedSystemAccessControlRules(
             @JsonProperty("catalogs") Optional<List<CatalogAccessControlRule>> catalogRules,
-            @JsonProperty("user_patterns") Optional<List<Pattern>> userPatterns)
+            @JsonProperty("principals") Optional<List<PrincipalUserMatchRule>> principalUserMatchRules)
     {
         this.catalogRules = catalogRules.map(ImmutableList::copyOf).orElse(ImmutableList.of());
-        this.userPatterns = userPatterns.map(ImmutableList::copyOf).orElse(ImmutableList.of());
+        this.principalUserMatchRules = principalUserMatchRules.map(ImmutableList::copyOf).orElse(ImmutableList.of());
```

Comment:
`principalUserMatchRules` could remain `Optional<List<..>>`

Reply:
I guess we should be better to keep the `catalogRules` and `principalUserMatchRules` consistant?

Reply:
Yea, consistency is good. However, no `catalogRules` mean access is denied. And here, no `principalUserMatchRules` rules mean access is allowed. So the semantics is inconsistent by design.

Reply:
Gotcha.
My understanding was that historically, there is a hack on the `system` catalog so that it's impossible that `catalogRules` is empty.